reactioncommerce/reaction-next-starterkit
# feat: Token-based login for dev

Ticket: #43 
Impact: **minor**
Type: **feat**

## Issue

Implement a token based authentication system

## Solution

- Added a login dropdown with an input field with a save button
- Used a Popover and omitted "Cancel" button as you cancel by clicking outside the popover.
- Added a mobx authStore for saving the token in state as well has saving / getting token from cookies
- Saved the token as a cookie and providing it to the client that way as well as storing it in mobx
- Didn't use Localstorage as it's redundant to the cookies
- Provide the token to apollo on both client and server
- Removed login token from env vars and various places where it was used from those vars

## Breaking

Login token must be supplied using the login dropdown.

## Testing

1. Start app
1. Use the Account icon in the navigation 
1. Enter a valid meteor login token and hit save
1. App should load with data
